src/main/java/com/google/devtools/build/lib/skyframe/ActionTemplateExpansionFunction.java

```diff
@@ -68,12 +71,13 @@ public SkyValue compute(SkyKey skyKey, Environment env)
     if (env.valuesMissing()) {
       return null;
     }
-    Iterable<TreeFileArtifact> inputTreeFileArtifacts = treeArtifactValue.getChildren();
+    List<TreeFileArtifact> inputTreeFileArtifacts = Lists.newArrayList(treeArtifactValue.getChildren());
```

Comment:
Is it feasible to try to sort even earlier? Also, does this have anything to do with the OS? getChildren() seems to be coming from a map anyway, I wouldn't have thought that's sorted?

Reply:
Nice catch, @tomlu . I think https://source.bazel.build/bazel/+/master:src/main/java/com/google/devtools/build/lib/skyframe/TreeArtifactValue.java;l=64?q=TreeArtifactValue might be a good place: use a TreeMap instead of a HashMap? That will also allow for Skyframe change pruning in case of cache hits.

Reply:
Alright, I will change that.

Reply:
@tomlu @janakdr I replaced the `Map` with an `SortedMap`, I still get the same sorted output :) - could you check if this is a better solution?

Thanks :)